### PR TITLE
Library libavif with tune butteraugli

### PR DESCRIPTION
**Pull request by Jamaika1**
Library libavif with new aom 2.0.2-8de6aba + tune butteraugli for jpeg + integrated jpegxl

### CODE REVIEW

File: src/codec_aom.c [hunk 1]
```diff
@@ -344,6 +344,9 @@ struct aomOptionDef
 static const struct aomOptionEnumList tuningEnum[] = { //
     { "psnr", AOM_TUNE_PSNR },                         //
     { "ssim", AOM_TUNE_SSIM },                         //
+#if CONFIG_TUNE_BUTTERAUGLI
```
Comment (wantehchang) [inline src/codec_aom.c:347]: Jamaika1: Thank you for the pull request. This test assumes the build system passes -DCONFIG_TUNE_BUTTERAUGLI=1 on the compiler command line. The official libavif build system doesn't do this.

Here is the libaom enum type in question:

```
typedef enum {
  AOM_TUNE_PSNR = 0,
  AOM_TUNE_SSIM = 1,
  /* NOTE: enums 2 and 3 unused */
  AOM_TUNE_VMAF_WITH_PREPROCESSING = 4,
  AOM_TUNE_VMAF_WITHOUT_PREPROCESSING = 5,
  AOM_TUNE_VMAF_MAX_GAIN = 6,
  AOM_TUNE_VMAF_NEG_MAX_GAIN = 7,
  AOM_TUNE_BUTTERAUGLI = 8,
} aom_tune_metric;
```

Ideally we want libaom's public headers to provide a macro that we can test for the presence of AOM_TUNE_BUTTERAUGLI in the aom_tune_metric enum type. I don't think this was done when AOM_TUNE_BUTTERAUGLI was added to the aom_tune_metric enum type. I will look into this.

An alternative is to use the numeric value of AOM_TUNE_BUTTERAUGLI instead, like so:

    { "butteraugli",  8},           // AOM_TUNE_BUTTERAUGLI = 8

**Review by wantehchang — CHANGES_REQUESTED** [review]

File: src/codec_aom.c (hunk 1 above)
Comment (Jamaika1) [reply src/codec_aom.c:347]: av1enc.exe -q 100 -420 -tune BUTTERAUGLI -size 1563x1558 -effort 5 -thread 4 image_21447_24bit.png -d image_21447_24bitx.av1
OK
test libwebp2 -> libavif -> aom
av1enc.exe -avif -q 100 -420 -tune BUTTERAUGLI -size 1563x1558 -effort 5 -thread 4 image_21447_24bit.png -d image_21447_24bitw.avif
```
ERROR: Failed to encode: Encoding of color planes failed
ERROR: Failed to encode: No AV1 items found
Error while computing the distortion.
```

Comment (joedrago) [reply src/codec_aom.c:347]: I'm unfamiliar with `av1enc` (it appears to be gstreamer). Does that use libavif internally if `-avif` is supplied on the commandline?

Comment (Jamaika1) [reply src/codec_aom.c:347]: This is the libwebp 2 program that uses libavif for google products.
I was unable to compile the avif file from libavif
https://www.sendspace.com/file/dzkvri
I was able to compile an avif file from libheif / avif 
https://github.com/strukturag/libheif/pull/448

**Comment by wantehchang** [comment]
This pull request has been replaced by https://github.com/AOMediaCodec/libavif/pull/567.

**Pull request closed without merging**